web-dapp/controllers/prepareRegTx.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+const {validate, normalize} = require('../server-lib/validations');
+const config = require('../server-config');
+const generateCode = require('../server-lib/generate_code');
+const recalcPrice = require('../server-lib/recalc_price');
+const buildSignature = require('../server-lib/buildSignature');
+const db = require('../server-lib/session_store');
+const postAPI = require('../server-lib/post_api');
+
+const signerPrivateKey = config.signer_private_key;
+
+const validateBody = (body) => {
```

Comment:
I suggest we remove these `validate...` functions and put this logic into `validateData`. It just does one thing, so having a lot of them make things looks more complicated than they actually are